### PR TITLE
Fix and simplify semantic-segmentation example

### PR DESCRIPTION
**Pull request by qubvel**
# What does this PR do?

It seems like the `semantic-segmentation` example is a bit outdated and doesn't run without fixing some issues. This PR does the following to improve the current state of the example:

1. Added instructions for accessing the sidewalk dataset to the semantic-segmentation example's README.md
2. Fixed missing dependencies required to run the semantic-segmentation example (accelerate, Pillow)
3. Fixed failing image transforms; replaced implemented transforms with those from `albumentations`
4. It was also observed that the "mean_iou" metric computation takes a lot of time and can be optimized. To improve this, I created a PR in the Evaluate project (huggingface/evaluate#569). As soon as the fix is merged, the time to perform training will be reduced by half.

Updated scripts have been evaluated and the fine-tuned model achieved similar metrics to the previous one:
 - previous model: https://huggingface.co/nielsr/segformer-finetuned-sidewalk-10k-steps
 - new model: https://huggingface.co/qubvel-hf/segformer-finetuned-sidewalk-10k-steps

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@amyeroberts @NielsRogge

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Very nice! Thanks for working on this and improving our examples!

File: examples/pytorch/semantic-segmentation/run_semantic_segmentation_no_trainer.py [hunk 1]
```diff
@@ -418,69 +317,59 @@ def main():
...
-            ReduceLabels() if args.reduce_labels else Identity(),
-            RandomCrop(size=size),
-            RandomHorizontalFlip(flip_prob=0.5),
-            PILToTensor(),
-            ConvertImageDtype(torch.float),
-            Normalize(mean=image_processor.image_mean, std=image_processor.image_std),
+            A.Lambda(name="reduce_labels", mask=reduce_labels_transform if args.reduce_labels else None, p=1.0),
+            A.PadIfNeeded(
+                min_height=height, min_width=width, border_mode=0, value=255, p=1.0
+            ),  # 255 is ignored by loss
```
[the remark this answers is not in the document]
Comment (qubvel) [reply examples/pytorch/semantic-segmentation/run_semantic_segmentation_no_trainer.py]: Thanks for the review, I fixed it in both files
